lib/spree_taxjar/categories.rb

```diff
@@ -0,0 +1,66 @@
+module SpreeTaxjar
+  class Categories
+    class << self
+      def update(categories = [])
+        categories = default_categories if categories.empty?
+        default_categories.each do |c|
+          p = convert_to_hash(c)
+          t = Spree::TaxCategory.where(name: name(p)).first
+          if t.present?
+            p "TaxCategory:: Update:: #{t.inspect}"
+            t.update_attributes!(transform(p))
+          else
+            new_category = Spree::TaxCategory.create!(transform(p))
+            p "TaxCategory:: Create:: #{new_category.inspect}"
+          end
+        end
+      end
+
+      def refresh
+        p "Taxjar:: Categories:: API Call started !!"
+        client = ::Taxjar::Client.new(api_key: Spree::Config[:taxjar_api_key])
+        categories = client.categories
+        p "Taxjar:: Categories:: Update Started"
+        update(categories)
+      end
+
+      private
+        def convert_to_hash(p)
+          p.to_h
+        end
+
+        def transform(p)
+          {name: name(p), tax_code: tax_code(p), description: description(p)}
+        end
+
+        def name(p)
+          p.fetch(:name)
+        end
+
+        def tax_code(p)
+          p.fetch(:product_tax_code)
+        end
+
+        def description(p)
+          p.fetch(:description)
+        end
+
+        def default_categories
+          [
+            {:name=>"Digital Goods", :product_tax_code=>"31000", :description=>"Digital products transferred electronically, meaning obtained by the purchaser by means other than tangible storage media."},
```

Comment:
@nishant-cyro can we skip the new hash syntax as this is generated via a script ( which fetches available product categories using taxjar api ) whose output is copied here and ruby prints the hash in rocket_syntax (=>) whether defined like json or rocket_syntax 😞